### PR TITLE
Temporary OCS API for user avatar upload and delete

### PR DESCRIPTION
**Pull request by nickvergessen**
### Upload an avatar
```
curl -k -H "OCS-APIRequest: true" -u admin:admin 'https://nextcloud22.local/ocs/v2.php/apps/spreed/temp-user-avatar' -X POST -F 'files[]=@/home/nickv/avatar.jpg'
```

### Delete an avatar
```
curl -k -H "OCS-APIRequest: true" -u admin:admin 'https://nextcloud22.local/ocs/v2.php/apps/spreed/temp-user-avatar' -X DELETE
```

Not adding documentation because it's being wiped soon again and will be removed without any close done time once the one from server is there.

### CODE REVIEW

**Comment by PVince81** [comment]
for the record, the real deal will be https://github.com/nextcloud/server/pull/24579

**Comment by PVince81** [comment]
Tested both endpoints and they worked fine with both jpg and png.

However, if you upload a non-square image there is no useful error message in the response:
```
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>400</statuscode>
  <message/>
 </meta>
 <data>
  <message>An error occurred. Please contact your admin.</message>
 </data>
</ocs>
```

I had to look at the nextcloud.log to see what the problem was.
As far as I saw in the code we wanted to copy the messages there. (could also be a bug in the old controller)

**Review by PVince81 — APPROVED** [review]
:+1: looks fine, see my previous comment about error messages

File: lib/Controller/TempAvatarController.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+					['data' => ['message' => $this->l->t('Unknown filetype')]],
+					Http::STATUS_BAD_REQUEST
+				);
+			}
+
+			$avatar = $this->avatarManager->getAvatar($this->userId);
+			$avatar->set($image);
+			return new DataResponse();
+		} catch (\Exception $e) {
+			$this->logger->error('Failed to delete avatar', [
```
Comment (nickvergessen) [inline lib/Controller/TempAvatarController.php:123]: this is post not delete, copy paste error in here

File: lib/Controller/TempAvatarController.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+				'exception' => $e,
+			]);
+
+			return new DataResponse(['message' => $this->l->t('An error occurred. Please contact your admin.')], Http::STATUS_BAD_REQUEST);
```
Comment (nickvergessen) [inline lib/Controller/TempAvatarController.php:127]: I guess we can forward the error from the avatar manager as well? That could solve the issue @PVince81 mentioned

**Comment by PVince81** [comment]
psalm not happy, might need to add to baseline as we're using the private `Filesystem` class for the blacklist check

**Comment by nickvergessen** [comment]
yeah, I bumped the baseline.

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**